src/libraries/Microsoft.Extensions.Logging.Console/src/TextWriterExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal static class TextWriterExtensions
+    {
+        public static bool WriteColoredMessage(this TextWriter textWriter, string message, ConsoleColor? background, ConsoleColor? foreground, bool disableColors)
+        {
+            if (disableColors)
+            {
+                textWriter.Write(message);
+                return false;
+            }
+            // Order: backgroundcolor, foregroundcolor, Message, reset foregroundcolor, reset backgroundcolor
+            if (background.HasValue)
+            {
+                textWriter.Write(AnsiParser.GetBackgroundColorEscapeCode(background.Value));
+            }
+            if (foreground.HasValue)
+            {
+                textWriter.Write(AnsiParser.GetForegroundColorEscapeCode(foreground.Value));
+            }
+            textWriter.Write(message);
+            if (foreground.HasValue)
+            {
+                textWriter.Write(AnsiParser.DefaultForegroundColor); // reset to default foreground color
+            }
+            if (background.HasValue)
+            {
+                textWriter.Write(AnsiParser.DefaultBackgroundColor); // reset to the background color
+            }
+            return background.HasValue || foreground.HasValue;
```

Comment:
This `return` value doesn't appear to be used. this can just be a `void` method.